Page flip animation

Hello:)
This feature request is making the page flip to give the user the experience of turning pages on a real book.

> > @Bashamega Is there any particular node version that should be used to install dependencies? Because I am not able to proceed further with "npm install"
> 
> It should work can you paste the error?

sagarhedaoo@Sagars-MBP ebookCraft % npm install
npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm WARN deprecated npmlog@6.0.2: This package is no longer supported.
npm WARN deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm WARN deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm WARN deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm WARN deprecated glob@7.1.7: Glob versions prior to v9 are no longer supported
npm WARN deprecated gauge@4.0.4: This package is no longer supported.
npm ERR! code 1
npm ERR! path /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass
npm ERR! command failed
npm ERR! command sh -c node scripts/build.js
npm ERR! Binary found at /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node
npm ERR! Testing binary
npm ERR! Binary has a problem: Error: dlopen(/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node, 0x0001): tried: '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node' (not a mach-o file), '/System/Volumes/Preboot/Cryptexes/OS/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node' (no such file), '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node' (not a mach-o file)
npm ERR!     at Module._extensions..node (node:internal/modules/cjs/loader:1340:18)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:1119:32)
npm ERR!     at Module._load (node:internal/modules/cjs/loader:960:12)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1143:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:110:18)
npm ERR!     at module.exports (/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/lib/binding.js:19:10)
npm ERR!     at Object.<anonymous> (/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/lib/index.js:13:35)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1256:14)
npm ERR!     at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:1119:32) {
npm ERR!   code: 'ERR_DLOPEN_FAILED'
npm ERR! }
npm ERR! Building the binary locally
npm ERR! Building: /Users/sagarhedaoo/.nvm/versions/node/v18.17.0/bin/node /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp verb cli [
npm ERR! gyp verb cli   '/Users/sagarhedaoo/.nvm/versions/node/v18.17.0/bin/node',
npm ERR! gyp verb cli   '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/bin/node-gyp.js',
npm ERR! gyp verb cli   'rebuild',
npm ERR! gyp verb cli   '--verbose',
npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! gyp verb cli   '--libsass_library='
npm ERR! gyp verb cli ]
npm ERR! gyp info using node-gyp@8.4.1
npm ERR! gyp info using node@18.17.0 | darwin | arm64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb find Python Python is not set from command line or npm configuration
npm ERR! gyp verb find Python Python is not set from environment variable PYTHON
npm ERR! gyp verb find Python checking if "python3" can be used
npm ERR! gyp verb find Python - executing "python3" to get executable path
npm ERR! gyp verb find Python - executable path is "/opt/homebrew/opt/python@3.12/bin/python3.12"
npm ERR! gyp verb find Python - executing "/opt/homebrew/opt/python@3.12/bin/python3.12" to get version
npm ERR! gyp verb find Python - version is "3.12.3"
npm ERR! gyp info find Python using Python version 3.12.3 found at "/opt/homebrew/opt/python@3.12/bin/python3.12"
npm ERR! gyp verb get node dir no --target version specified, falling back to host node version: 18.17.0
npm ERR! gyp verb command install [ '18.17.0' ]
npm ERR! gyp verb install input version string "18.17.0"
npm ERR! gyp verb install installing version: 18.17.0
npm ERR! gyp verb install --ensure was passed, so won't reinstall if already installed
npm ERR! gyp verb install version is already installed, need to check "installVersion"
npm ERR! gyp verb got "installVersion" 9
npm ERR! gyp verb needs "installVersion" 9
npm ERR! gyp verb install version is good
npm ERR! gyp verb get node dir target node version installed: 18.17.0
npm ERR! gyp verb build dir attempting to create "build" dir: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/build
npm ERR! gyp verb build dir "build" dir needed to be created? Yes
npm ERR! gyp verb build/config.gypi creating config file
npm ERR! gyp verb build/config.gypi writing out config file: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/build/config.gypi
npm ERR! gyp verb config.gypi checking for gypi file: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/config.gypi
npm ERR! gyp verb common.gypi checking for gypi file: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/common.gypi
npm ERR! gyp verb gyp gyp format was not specified; forcing "make"
npm ERR! gyp info spawn /opt/homebrew/opt/python@3.12/bin/python3.12
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/sagarhedaoo/Library/Caches/node-gyp/18.17.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/sagarhedaoo/Library/Caches/node-gyp/18.17.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/sagarhedaoo/Library/Caches/node-gyp/18.17.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! Traceback (most recent call last):
npm ERR!   File "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/gyp_main.py", line 42, in <module>
npm ERR!     import gyp  # noqa: E402
npm ERR!     ^^^^^^^^^^
npm ERR!   File "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 9, in <module>
npm ERR!     import gyp.input
npm ERR!   File "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 19, in <module>
npm ERR!     from distutils.version import StrictVersion
npm ERR! ModuleNotFoundError: No module named 'distutils'
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/lib/configure.js:259:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Darwin 23.5.0
npm ERR! gyp ERR! command "/Users/sagarhedaoo/.nvm/versions/node/v18.17.0/bin/node" "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass
npm ERR! gyp ERR! node -v v18.17.0
npm ERR! gyp ERR! node-gyp -v v8.4.1
npm ERR! gyp ERR! not ok 
npm ERR! Build failed with error code: 1

> > > @Bashamega Is there any particular node version that should be used to install dependencies? Because I am not able to proceed further with "npm install"
> > 
> > 
> > It should work can you paste the error?
> 
> sagarhedaoo@Sagars-MBP ebookCraft % npm install npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful. npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs npm WARN deprecated npmlog@6.0.2: This package is no longer supported. npm WARN deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported npm WARN deprecated are-we-there-yet@3.0.1: This package is no longer supported. npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported npm WARN deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported npm WARN deprecated glob@7.1.7: Glob versions prior to v9 are no longer supported npm WARN deprecated gauge@4.0.4: This package is no longer supported. npm ERR! code 1 npm ERR! path /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass npm ERR! command failed npm ERR! command sh -c node scripts/build.js npm ERR! Binary found at /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node npm ERR! Testing binary npm ERR! Binary has a problem: Error: dlopen(/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node, 0x0001): tried: '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node' (not a mach-o file), '/System/Volumes/Preboot/Cryptexes/OS/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node' (no such file), '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/vendor/darwin-arm64-108/binding.node' (not a mach-o file) npm ERR! at Module._extensions..node (node:internal/modules/cjs/loader:1340:18) npm ERR! at Module.load (node:internal/modules/cjs/loader:1119:32) npm ERR! at Module._load (node:internal/modules/cjs/loader:960:12) npm ERR! at Module.require (node:internal/modules/cjs/loader:1143:19) npm ERR! at require (node:internal/modules/cjs/helpers:110:18) npm ERR! at module.exports (/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/lib/binding.js:19:10) npm ERR! at Object. (/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/lib/index.js:13:35) npm ERR! at Module._compile (node:internal/modules/cjs/loader:1256:14) npm ERR! at Module._extensions..js (node:internal/modules/cjs/loader:1310:10) npm ERR! at Module.load (node:internal/modules/cjs/loader:1119:32) { npm ERR! code: 'ERR_DLOPEN_FAILED' npm ERR! } npm ERR! Building the binary locally npm ERR! Building: /Users/sagarhedaoo/.nvm/versions/node/v18.17.0/bin/node /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library= npm ERR! gyp info it worked if it ends with ok npm ERR! gyp verb cli [ npm ERR! gyp verb cli '/Users/sagarhedaoo/.nvm/versions/node/v18.17.0/bin/node', npm ERR! gyp verb cli '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/bin/node-gyp.js', npm ERR! gyp verb cli 'rebuild', npm ERR! gyp verb cli '--verbose', npm ERR! gyp verb cli '--libsass_ext=', npm ERR! gyp verb cli '--libsass_cflags=', npm ERR! gyp verb cli '--libsass_ldflags=', npm ERR! gyp verb cli '--libsass_library=' npm ERR! gyp verb cli ] npm ERR! gyp info using node-gyp@8.4.1 npm ERR! gyp info using node@18.17.0 | darwin | arm64 npm ERR! gyp verb command rebuild [] npm ERR! gyp verb command clean [] npm ERR! gyp verb clean removing "build" directory npm ERR! gyp verb command configure [] npm ERR! gyp verb find Python Python is not set from command line or npm configuration npm ERR! gyp verb find Python Python is not set from environment variable PYTHON npm ERR! gyp verb find Python checking if "python3" can be used npm ERR! gyp verb find Python - executing "python3" to get executable path npm ERR! gyp verb find Python - executable path is "/opt/homebrew/opt/python@3.12/bin/python3.12" npm ERR! gyp verb find Python - executing "/opt/homebrew/opt/python@3.12/bin/python3.12" to get version npm ERR! gyp verb find Python - version is "3.12.3" npm ERR! gyp info find Python using Python version 3.12.3 found at "/opt/homebrew/opt/python@3.12/bin/python3.12" npm ERR! gyp verb get node dir no --target version specified, falling back to host node version: 18.17.0 npm ERR! gyp verb command install [ '18.17.0' ] npm ERR! gyp verb install input version string "18.17.0" npm ERR! gyp verb install installing version: 18.17.0 npm ERR! gyp verb install --ensure was passed, so won't reinstall if already installed npm ERR! gyp verb install version is already installed, need to check "installVersion" npm ERR! gyp verb got "installVersion" 9 npm ERR! gyp verb needs "installVersion" 9 npm ERR! gyp verb install version is good npm ERR! gyp verb get node dir target node version installed: 18.17.0 npm ERR! gyp verb build dir attempting to create "build" dir: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/build npm ERR! gyp verb build dir "build" dir needed to be created? Yes npm ERR! gyp verb build/config.gypi creating config file npm ERR! gyp verb build/config.gypi writing out config file: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/build/config.gypi npm ERR! gyp verb config.gypi checking for gypi file: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/config.gypi npm ERR! gyp verb common.gypi checking for gypi file: /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/common.gypi npm ERR! gyp verb gyp gyp format was not specified; forcing "make" npm ERR! gyp info spawn /opt/homebrew/opt/python@3.12/bin/python3.12 npm ERR! gyp info spawn args [ npm ERR! gyp info spawn args '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/gyp_main.py', npm ERR! gyp info spawn args 'binding.gyp', npm ERR! gyp info spawn args '-f', npm ERR! gyp info spawn args 'make', npm ERR! gyp info spawn args '-I', npm ERR! gyp info spawn args '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass/build/config.gypi', npm ERR! gyp info spawn args '-I', npm ERR! gyp info spawn args '/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/addon.gypi', npm ERR! gyp info spawn args '-I', npm ERR! gyp info spawn args '/Users/sagarhedaoo/Library/Caches/node-gyp/18.17.0/include/node/common.gypi', npm ERR! gyp info spawn args '-Dlibrary=shared_library', npm ERR! gyp info spawn args '-Dvisibility=default', npm ERR! gyp info spawn args '-Dnode_root_dir=/Users/sagarhedaoo/Library/Caches/node-gyp/18.17.0', npm ERR! gyp info spawn args '-Dnode_gyp_dir=/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp', npm ERR! gyp info spawn args '-Dnode_lib_file=/Users/sagarhedaoo/Library/Caches/node-gyp/18.17.0/<(target_arch)/node.lib', npm ERR! gyp info spawn args '-Dmodule_root_dir=/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass', npm ERR! gyp info spawn args '-Dnode_engine=v8', npm ERR! gyp info spawn args '--depth=.', npm ERR! gyp info spawn args '--no-parallel', npm ERR! gyp info spawn args '--generator-output', npm ERR! gyp info spawn args 'build', npm ERR! gyp info spawn args '-Goutput_dir=.' npm ERR! gyp info spawn args ] npm ERR! Traceback (most recent call last): npm ERR! File "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/gyp_main.py", line 42, in npm ERR! import gyp # noqa: E402 npm ERR! ^^^^^^^^^^ npm ERR! File "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/pylib/gyp/**init**.py", line 9, in npm ERR! import gyp.input npm ERR! File "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 19, in npm ERR! from distutils.version import StrictVersion npm ERR! ModuleNotFoundError: No module named 'distutils' npm ERR! gyp ERR! configure error npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1 npm ERR! gyp ERR! stack at ChildProcess.onCpExit (/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/lib/configure.js:259:16) npm ERR! gyp ERR! stack at ChildProcess.emit (node:events:514:28) npm ERR! gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:291:12) npm ERR! gyp ERR! System Darwin 23.5.0 npm ERR! gyp ERR! command "/Users/sagarhedaoo/.nvm/versions/node/v18.17.0/bin/node" "/Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library=" npm ERR! gyp ERR! cwd /Users/sagarhedaoo/Downloads/ebookCraft/node_modules/node-sass npm ERR! gyp ERR! node -v v18.17.0 npm ERR! gyp ERR! node-gyp -v v8.4.1 npm ERR! gyp ERR! not ok npm ERR! Build failed with error code: 1

I think the problem is with your setup, because i have tried it and it worked. I also works on gitpod

--- a/src/components/book.jsx
+++ b/src/components/book.jsx
@@ -1,88 +1,110 @@
 import React, { useState, useEffect } from "react";
-import { BiChevronLeft } from "react-icons/bi";
-import { BiChevronRight } from "react-icons/bi";
+import { BiChevronLeft, BiChevronRight } from "react-icons/bi";
 import settings from "@/setting/settings.json";
+import styles from "../styles/Book.module.css";
 
 export function Book() {
   const [currentPage, setCurrentPage] = useState("default.html");
   const [pageContent, setPageContent] = useState("");
-  const [current, setCurrent] = useState(1)
-  const [pages, setpages] = useState([])
+  const [current, setCurrent] = useState(1);
+  const [pages, setPages] = useState([]);
+  const [animationClass, setAnimationClass] = useState("");
+
   useEffect(() => {
-    setCurrentPage(pages[current-1])
-    console.log(pages)
-    console.log(pages.length)
-    if(currentPage){
+    setCurrentPage(pages[current - 1]);
+    console.log(pages);
+    console.log(pages.length);
+    if (currentPage) {
       const pagePath = `/pages/${currentPage}`;
-    fetch(pagePath)
-      .then((response) => response.text())
-      .then((data) => setPageContent(data))
-      .catch((error) => console.error("Error fetching page content:", error));
-    switch (current) {
-      case 1:
-        document.getElementById('back').classList.add('disabled')
-        document.getElementById('forward').classList.remove('disabled')
-        break;
-      case pages.length:
-        document.getElementById('forward').classList.add('disabled')
-        document.getElementById('back').classList.remove('disabled')
-        break;
-      default:
-        document.getElementById('back').classList.remove('disabled')
-        document.getElementById('forward').classList.remove('disabled')
+      fetch(pagePath)
+        .then((response) => response.text())
+        .then((data) => setPageContent(data))
+        .catch((error) => console.error("Error fetching page content:", error));
 
-        break;
-    }
-    
-    }
-  }, [currentPage, current, pages]);
-  useEffect(()=>{
-    
-    if(settings.pageSaver){
-      if(localStorage.getItem(settings.title + "_pagesaver") !== null){
-        const page = parseInt(localStorage.getItem(settings.title + "_pagesaver"))
-        setCurrent(page)
-
+      switch (current) {
+        case 1:
+          document.getElementById("back").classList.add("disabled");
+          document.getElementById("forward").classList.remove("disabled");
+          break;
+        case pages.length:
+          document.getElementById("forward").classList.add("disabled");
+          document.getElementById("back").classList.remove("disabled");
+          break;
+        default:
+          document.getElementById("back").classList.remove("disabled");
+          document.getElementById("forward").classList.remove("disabled");
+          break;
       }
     }
-    fetch("/api/pagesRender").then(res=>res.json()).then(data=>setpages(data))
-  }, [])
-  const back = ()=>{
-    if(!document.getElementById('back').classList.contains('disabled')) {
-      setCurrent((prevCurrent) => prevCurrent - 1);
-      localStorage.setItem(settings.title + "_pagesaver", current - 1)
+  }, [currentPage, current, pages]);
 
+  useEffect(() => {
+    if (settings.pageSaver) {
+      if (localStorage.getItem(settings.title + "_pagesaver") !== null) {
+        const page = parseInt(localStorage.getItem(settings.title + "_pagesaver"));
+        setCurrent(page);
+      }
     }
-    
-  }
-  const forward = ()=>{
-    if(!document.getElementById('forward').classList.contains('disabled')) {
-      setCurrent((prevCurrent) => prevCurrent + 1);
-      localStorage.setItem(settings.title + "_pagesaver", current + 1)
+    fetch("/api/pagesRender")
+      .then((res) => res.json())
+      .then((data) => setPages(data));
+  }, []);
+
+  const back = () => {
+    if (!document.getElementById("back").classList.contains("disabled")) {
+      setAnimationClass(styles.iframeSlideOutRight);
+      setTimeout(() => {
+        setCurrent((prevCurrent) => prevCurrent - 1);
+        localStorage.setItem(settings.title + "_pagesaver", current - 1);
+        setAnimationClass(styles.iframeSlideInLeft);
+      }, 500);
     }
-  }
+  };
+
+  const forward = () => {
+    if (!document.getElementById("forward").classList.contains("disabled")) {
+      setAnimationClass(styles.iframeSlideOutLeft);
+      setTimeout(() => {
+        setCurrent((prevCurrent) => prevCurrent + 1);
+        localStorage.setItem(settings.title + "_pagesaver", current + 1);
+        setAnimationClass(styles.iframeSlideInRight);
+      }, 500);
+    }
+  };
+
   return (
     <main id="book">
-      <iframe title="book-page" srcDoc={pageContent}></iframe>
+      <div className={`${styles.iframeContainer} ${animationClass}`}>
+        <iframe title="book-page" srcDoc={pageContent}></iframe>
+      </div>
       <footer>
-        <span id="forward" onClick={forward}>
+        <span id="back" onClick={back}>
           <BiChevronLeft />
         </span>
         <div>
-        <h2 onClick={() => setCurrentPage('default.html')}>{settings.title}</h2>
-        <p>Created by {settings.author}</p>
-        <p>{pages.length}/
-        <input type="number" min="1" max={pages.length} onChange={(e)=>{
-          if(e.target.value <= 0 || e.target.value > pages.length){
-            e.target.classList.add('error')
-          }else{
-            e.target.classList.remove('error')
-            setCurrent(e.target.value)
-          }
-        }} value={current} style={{width: '50px'}}></input>
-        </p>
+          <h2 onClick={() => setCurrentPage("default.html")}>{settings.title}</h2>
+          <p>Created by {settings.author}</p>
+          <p>
+            {pages.length}/
+            <input
+              type="number"
+              min="1"
+              max={pages.length}
+              onChange={(e) => {
+                const newValue = parseInt(e.target.value);
+                if (newValue <= 0 || newValue > pages.length) {
+                  e.target.classList.add("error");
+                } else {
+                  e.target.classList.remove("error");
+                  setCurrent(newValue);
+                }
+              }}
+              value={current}
+              style={{ width: "50px" }}
+            ></input>
+          </p>
         </div>
-        <span id="back" onClick={back}>
+        <span id="forward" onClick={forward}>
           <BiChevronRight />
         </span>
       </footer>

--- a/src/styles/Book.module.css
+++ b/src/styles/Book.module.css
@@ -1,0 +1,67 @@
+/* src/styles/Book.module.css */
+
+/* Define keyframes for slide animations */
+@keyframes slideInFromLeft {
+  from {
+    transform: translateX(-100%);
+    opacity: 0;
+  }
+  to {
+    transform: translateX(0);
+    opacity: 1;
+  }
+}
+
+@keyframes slideInFromRight {
+  from {
+    transform: translateX(100%);
+    opacity: 0;
+  }
+  to {
+    transform: translateX(0);
+    opacity: 1;
+  }
+}
+
+@keyframes slideOutToLeft {
+  from {
+    transform: translateX(0);
+    opacity: 1;
+  }
+  to {
+    transform: translateX(-100%);
+    opacity: 0;
+  }
+}
+
+@keyframes slideOutToRight {
+  from {
+    transform: translateX(0);
+    opacity: 1;
+  }
+  to {
+    transform: translateX(100%);
+    opacity: 0;
+  }
+}
+
+/* Apply animation to the iframe */
+.iframeContainer {
+  position: relative;
+}
+
+.iframeSlideInLeft {
+  animation: slideInFromLeft 0.5s forwards;
+}
+
+.iframeSlideInRight {
+  animation: slideInFromRight 0.5s forwards;
+}
+
+.iframeSlideOutLeft {
+  animation: slideOutToLeft 0.5s forwards;
+}
+
+.iframeSlideOutRight {
+  animation: slideOutToRight 0.5s forwards;
+}

